src/pybind/mgr/dashboard/frontend/src/app/core/auth/user-form/user-form.component.html

```diff
@@ -23,13 +23,19 @@
                    name="username"
                    formControlName="username"
                    autocomplete="off"
-                   autofocus>
+                   autofocus
+                   ngbTooltip="Empty spaces will be trimmed"
+                   i18n-ngbTooltip
+                   (blur)="onBlurTrim('username')">
```

Comment:
and then you won't need an extra function to trim it. Ideally there should be a pipe that does this for us so we don't need to do this on every component that we want to trim the whitespace.
```suggestion
                   i18n-ngbTooltip
                   #usernameInput
                   (blur)="usernameInput.value = usernameInput.value.trim()">
```

Reply:
There is the `cdTrim` directive that does the job, so using that one